js/components/moter/motebehov/SvarMotebehovSkjema.js

```diff
@@ -0,0 +1,245 @@
+    {
+        felt,
+        handleOptionChange,
+    }) => {
+    return (
+        <div className="skjemaelement">
+            <h3
+                className="skjemaelement__sporsmal"
+                id={felt.navn}
+            >
+                {felt.spoersmaal}
+            </h3>
+            <Field
+                autoFocus
+                id={felt.navn}
+                name={felt.navn}
+                component={Radioknapper}
+                onChange={(event) => {
+                    handleOptionChange(event.target.value);
+                }}
+            >
+                {
+                    felt.svar.map((svar, index) => {
+                        return (
+                            <input
+                                key={`vilHaMote-${index}`}
+                                value={svar.verdi}
+                                label={svar.tekst}
+                                id={`${felt.navn}-${index}`}
+                                aria-labelledby={felt.navn}
+                            />
+                        );
+                    })
+                }
+            </Field>
+        </div>
+    );
+};
+VilHaMoteSvarKnapper.propTypes = {
+    felt: felterPt,
+    handleOptionChange: PropTypes.func,
+};
+
+export const MotebehovSkjemaTekstomraade = (
+    {
+        felt,
+    }) => {
+    return (
+        <div className="skjema_element motebehovSkjemaTekstomraade">
+            <h3
+                className="skjemaelement__sporsmal"
+                id={felt.navn}
+            >
+                {felt.spoersmaal}
+            </h3>
+            <TekstSensitiv />
+            <Field
+                className="input--fullbredde"
+                name={felt.navn}
+                id={`${felt.navn}-input`}
+                aria-labelledby={felt.navn}
+                component={Tekstomraade}
+                placeholder={'Skriv her'}
+                rows="5"
+            />
+        </div>
+    );
+};
+MotebehovSkjemaTekstomraade.propTypes = {
+    felt: felterPt,
+};
+
+export const TekstSensitiv = () => {
+    return (
+        <div className="svarMotebehovSkjema__tekstSensitiv">
+            {TEKSTER_INFORMASJON.sensitiv}
+        </div>
+    );
+};
+export const TekstOpplysning = () => {
+    const TEKSTER = {
+        tekstOpplysning: {
+            tekst: 'Vi bruker opplysningene du gir oss til å forbedre tjenestene våre. ',
+            lenke: 'Les mer om hvordan NAV behandler personopplysninger.',
+        },
+    };
+    return (
+        <div className="svarMotebehovSkjema__tekstOpplysning">
+            <p>
+                {TEKSTER.tekstOpplysning.tekst}<a
+                    className="lenke"
+                    href="http://www.nav.no/personvern"
+                    title="Følg lenke">{TEKSTER.tekstOpplysning.lenke}</a>
+            </p>
+        </div>
+    );
+};
+
+export class SvarMotebehovSkjemaKomponent extends Component {
+    constructor(props) {
```

Comment:
Tanken med redux-form er at tilstanden til skjemaet skal ligge i redux-store. Dette får du ut av boksen. Du bør med andre ord ikke legge på noe changehandlers på input-elementene med mindre du har helt spesielle behov (noe du ikke har her). Les i stedet ut verdien med mapStateToProps og redux-form sin `getFormValues`-selector (se dokumentasjon, eller evt. `js/components/sykepengesoknad-utland/UtlandsSkjema/UtlandsSkjema.js`

Kan ellers tipse om at Barken har laget en del skjemaelementer som kan brukes til det behovet du har, eksempelvis JaEllerNei som animerer opp et underspørsmål dersom man svarer ja (eller nei), eller en mer generisk variant av denne som heter SporsmalMedTillegg. Ville ha brukt disse komponentene i stedet.

Skjemaet mangler FeilOppsummering. Se f.eks. `js/components/sykepengesoknad-utland/UtlandsSkjema/UtlandsSkjema.js` for hvordan dette lages.

Reply:
Har ordnet med getFormValues, men er vel litt overflødig med FeilOppsummering og det er max 1 feil som oppstå med 1 felt.